src/txmempool.cpp

```diff
@@ -447,6 +450,15 @@ void CTxMemPool::removeUnchecked(txiter it)
     BOOST_FOREACH(const CTxIn& txin, it->GetTx().vin)
         mapNextTx.erase(txin.prevout);
 
+    if (vTxHashes.size() > 1) {
+        vTxHashes[it->vTxHashesIdx] = vTxHashes[vTxHashes.size() - 1];
+        txiter newhashit = mapTx.find(vTxHashes[it->vTxHashesIdx].first);
```

Comment:
They are never invalidated until you delete the item itself.
http://www.boost.org/doc/libs/1_60_0/libs/multi_index/doc/tutorial/indices.html#guarantees